### ISSUE TITLE
item_cov and item_cov_miss

### ISSUE THREAD

**Issue by michalovadek** [saudiwin/idealstan#32]
if I understand it correctly, `item_cov=` and `item_cov_miss=` are used to specify the respective covariate adjustment for the regular and missing value models. But what happens if there is overlap among them in terms of some of the covariates?

On the implementation side there seems to be an issue. Specifically, if the same covariate appears in both formulas, the renaming functions try to create two copies of the same column which results in automatic renaming like so:

```r
`decade_init2000s` -> `decade_init2000s...21`
`decade_init2010s` -> `decade_init2010s...22`
```
Downstream this prevents `id_estimate` to identify the columns.

Should the handling of columns in `id_make` be adjusted if both `item_cov=` and `item_cov_miss=` are specified?

**Comment by saudiwin** [comment]
this could be an issue, although my preferred way to deal with it for now would be to force people to only use one type of covariate at a time (e.g., person/item etc).

**Comment by michalovadek** [comment]
well in this specific example the covariate is at the item level only. The issue arises from it appearing in both formulas

**Comment by saudiwin** [comment]
Yes I see the problem, I meant having the user specify only one of person_cov, item_cov, or item_miss_cov would make the error go away for now.

It wouldn't be hard in a larger inference sense to have them use covariates for different types of parameters. The main trick is with coding, we would need to append some prefix like "person_" or "item_" to the covariate matrix and then remember with helping functions to convert it. So definitely doable, just not high on the to-do list atm.

**Comment by saudiwin** [comment]
I think the way to deal with this is to preprend the covariate type (item, item_miss, person) to the covariate matrix when it is added to the idealdata object.
